src/graph-layouts.jl

```diff
@@ -0,0 +1,89 @@
+export distgraph, distdigraph, num_vertices, vertices, adj
+###
+# Graph data structure definitions.
+###
+
+# Allowed Graph datastructures:
+typealias AdjacencyList Array{Array{Int, 1}, 1}
+typealias AdjacencyMatrix AbstractArray{Int, 2}                      # Covers sparse matrices
+typealias GraphStruct Union{AdjacencyList, AdjacencyMatrix}
+
+
+# Generic (undirected or directed, up to the user to decide) Graph
+type DistGraph{S<:GraphStruct}
+    nv::Int                            # not really required, but kept for convenience
+    vertices::UnitRange{Int}
```

Comment:
The vertices field denotes the range of vertices present locally. It allows the workers to determine the global index of the first worker they have. But I guess the metadata will do something similar.